agent/plugin/definition_test.go

```diff
@@ -131,9 +132,15 @@ func TestDefinitionWithoutAdditionalProperties(t *testing.T) {
 	if res.Valid() {
 		t.Errorf("validator.Validate(def, {llamas:always,camels:never}).Valid() = true, want false")
 	}
+
 	// TODO: Testing error strings is fragile - replace with a more semantic test.
-	if got, want := res.Error(), `/camels: "never" cannot match schema`; got != want {
-		t.Errorf("validator.Validate(def, {llamas:always,camels:never}).Error() = %q, want %q", got, want)
+	errStr := res.Error()
+	if errStr == "" {
+		t.Errorf("expected validator.Validate(def, {llamas:always,camels:never}).Error() to be non-empty")
```

Comment:
I know this was here before this PR, but what do `llamas` have to do with the test data? I think in this case its 
```
			"properties": {
				"alpacas": {
					"type": "string"
				}
			},
```
from line 117.

I think we should have simpler error messages that won't be misleading if the test data changes.